File: skills/javascript_basic/notes_store.js
class NotesStore {
  constructor() {
    this.noteCollection = {};
  }

  addNote(state, name) {
    if (name === "") {
      throw new Error("Name cannot be empty");
    } else if (
      state !== "completed" &&
      state !== "active" &&
      state !== "others"
    ) {
      throw new Error(`Invalid state ${state}`);
    } else {
      this.noteCollection[name] = state;
    }
  }

  getNotes(state) {
    const notesList = [];

    if (state !== "completed" && state !== "active" && state !== "others") {
      throw new Error(`Invalid state ${state}`);
    } else {
      for (const [key, value] of Object.entries(this.noteCollection)) {
        if (state === value) {
          notesList.push(key);
        }
      }
    }

    return notesList;
  }
}

File: skills/c#_basic/Notes_Store.cs
public class NotesStore
{
    public IDictionary<string,string> noteCollection = new Dictionary<string, string>();
    
    public NotesStore() {}
    public void AddNote(String state, String name) {
        if(name == ""){
            throw new Exception("Name cannot be empty");
        }else if(state != "completed" && state != "active" && state != "others")
        {
            throw new Exception($"Invalid state {state}");
        }else
        {
            noteCollection.Add(name, state);
        }
        
    }
    public List<String> GetNotes(String state) {
        List<string> li = new List<string>();
        if(state != "completed" && state != "active" && state != "others")
        {
            throw new Exception($"Invalid state {state}");
        }else
        {
            foreach(KeyValuePair<string, string> kvp in noteCollection)
            {
                if(state == kvp.Value){
                    li.Add(kvp.Key);
                }
            }
        }
        return li;
        
    }
} 
